### PR TITLE
feat: added single product page

### PR DESCRIPTION
**Pull request by Chakravarthi0**

### CODE REVIEW

File: src/components/single-product-card/SingleProductCard.jsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
+import React, { useState } from "react";
+import { useCart, useAuth, useWishlist } from "../../context/index";
```
Comment (manojasarsa) [inline src/components/single-product-card/SingleProductCard.jsx:2]: You can give the relative path till ../../context - it will still work

**Review by manojasarsa — APPROVED** [review]
The code is pretty clean.
Good to merge!

File: src/components/single-product-card/SingleProductCard.jsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  const [isLoading, setIsLoading] = useState({
+    cart: false,
+    wishlist: false,
+  });
+
+  const navigate = useNavigate();
+
+  const { cartItems, addToCart } = useCart();
+  const { wishlistItems, addToWishlist, removeFromWishlist } = useWishlist();
+  const isInWishlist = wishlistItems.find((item) => item._id === product?._id);
```
Comment (kshitijthatte) [inline src/components/single-product-card/SingleProductCard.jsx:16]: You're using `isInWishlist` in multiple places (here and in `ProductCard.jsx`), so you can export it and make it a helper function.

File: src/components/single-product-card/SingleProductCard.jsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                <span
+                  className="rated"
+                  style={{ "--stars": product?.rating }}
+                ></span>
+                <span className="text-large">(300 reviews)</span>
+                <p>inclusive of all taxes</p>
+              </div>
+              <div className="product-details-btns">
+                {token ? (
+                  cartItems.find((item) => item._id === product?._id) ? (
```
Comment (kshitijthatte) [inline src/components/single-product-card/SingleProductCard.jsx:58]: You can make this a function, like you did with `isInWishlist`

**Review by kshitijthatte — APPROVED** [review]
Good work!

File: src/components/single-product-card/SingleProductCard.jsx (hunk 2 above)
Comment (Chakravarthi0) [reply src/components/single-product-card/SingleProductCard.jsx:16]: Thanks for the suggestion, I will do that while refactoting.

**Pull request merged**